mipengine/algorithms/specifications.py

```diff
@@ -184,17 +214,9 @@ def _validate_parameter_type_dict_enums(param_value, cls_values):
     _validate_parameter_type_dict_enums_not_allowed(param_value, cls_values)
 
 
-class AlgorithmSpecification(ImmutableBaseModel):
-    name: str
-    desc: str
-    label: str
-    enabled: bool
-    inputdata: InputDataSpecifications
-    parameters: Optional[Dict[str, ParameterSpecification]]
-    flags: Optional[Dict[str, bool]]
-
+class WorkflowStepSpecification(ImmutableBaseModel):
```

Comment:
I disagree with Jason's point of view, I also think that `Algorithm` and `Transformer` classes should inherit from a mother class that will contain the `get_specification`, `run` and maybe the `get_name` methods. This is though a discussion for a different time.

For the points that you made:
I am not sure how my implementation creates more duplication. It could be better, yes but it's not worse than adding the `validate_parameters` method into both classes. I agree that this could be improved in the future with more common parameters but I didn't want to go into this process so I improved only the relevant to this PR parts. 

The problem with the `Specification` name is that there are many specifications in this file, not just the Algorithm and the Transformer, so why isn't all of them inheriting from it. 

From what I could read the workflow in ML is pretty much what we do here, the term is a lot of times identical to the `Pipeline` term, but I think that a workflow is more common in systems as well. We could change it in the future together with the other discussion.

To summarize, I wanted to focus on this PR, without opening different topics, but at the same time improve small parts that were easy to do.